src/tools/compiletest/src/runtest.rs

```diff
@@ -1349,8 +1349,10 @@ impl<'test> TestCx<'test> {
         };
         rustc.arg(input_file);
 
-        // Use a single thread for efficiency and a deterministic error message order
-        rustc.arg("-Zthreads=1");
+        if !self.props.parallel_front_end {
+            // Use a single thread for efficiency and a deterministic error message order
+            rustc.arg("-Zthreads=1");
+        }
```

Comment:
Discussion: so my immediate reaction (when reading this part specifically, have not looked at other parts yet) is: do we ever want to support a custom specified fixed amounts of threads?

Reply:
This change is based on the assumption that we will soon stabilize the parallel front end and use a thread number greater than 1 by default. But at present we still need to solve many problems, and the compiler will still keep the thread number 1 by default in the foreseeable time. So I suggest deleting this code

Reply:
I think that's fair.